src/Forms/GridField/GridFieldFilterHeader.php

```diff
@@ -208,9 +234,15 @@ public function canFilterAnyColumns($gridField)
      */
     public function getSearchContext(GridField $gridField)
     {
-        $context = singleton($gridField->getModelClass())->getDefaultSearchContext();
+        if (!$this->searchContext) {
+            $this->searchContext = singleton($gridField->getModelClass())->getDefaultSearchContext();
+        }
 
-        return $context;
+        if ($this->updateSearchContextCallback) {
```

Comment:
I think it would be safer to move this condition into the prior one so this context is only likely to be called once when the context is first created.

Reply:
Yep, done